Source/Tools/run_vi_analyzer.py

```diff
@@ -0,0 +1,38 @@
+import logging
+import os
+import subprocess
+import sys
+import tempfile
+
+_logger = logging.getLogger(__name__)
+_logger.setLevel(logging.DEBUG)
+
+handler = logging.StreamHandler(sys.stdout)
+handler.setLevel(logging.DEBUG)
+_logger.addHandler(handler)
+
+def main():
+    return_code = run_all_tests()
+    sys.exit(return_code)
+
+
+def run_all_tests():
+    tools_directory = os.path.abspath(os.path.dirname(__file__))
+    vi_analyzer_config_path = os.path.join(tools_directory, "PR.viancfg")
+    _logger.debug(f"Analyzing VIs per {vi_analyzer_config_path}.")
+    vi_analyzer_output_path = tempfile.TemporaryFile().name
+    kwargs = ["LabVIEWCLI", "-OperationName", "RunVIAnalyzer", "-ConfigPath", vi_analyzer_config_path, "-ReportPath", vi_analyzer_output_path]
+    test_result = subprocess.run(kwargs, capture_output= True)
+    
+    formatted_stdout = test_result.stdout.decode().replace('\r\n','\n').strip()
+    _logger.debug(formatted_stdout)
+    if(test_result.returncode != 0):
+        formatted_stderr = test_result.stderr.decode().replace('\r\n','\n').strip()
+        _logger.error(formatted_stderr)
```

Comment:
Is this what we would look at to debug any failures? Do you have an example of what this looks like?

Reply:
Yes!  Shown below is an example of failures pertaining to overlapping diagram content, which will be viewable from the workflow failure log:

```
VI Analyzer Results

Original Analysis Performed	Wednesday, December 13, 2023 4:08:00 PM
Total Analysis Time	00:02:53.22

Results
VIs Analyzed	779
Total Tests Run	13243
Passed Tests	13242
Failed Tests	1
Skipped Tests	0

Errors
VI not loadable	0
Test not loadable	0
Test not runnable	0
Test error out	0

Failed Tests (sorted by VI)

Start Discovery Service.vi (C:\github\measurementlink-labview\Source\Runtime\MeasurementLink Discovery V1\Start Discovery Service.vi)
Hidden Objects in Structures	This constant is not entirely within the visible bounds of its owner (Case Structure, "False" case).
Hidden Objects in Structures	This constant is not entirely within the visible bounds of its owner (Case Structure, "False" case).
Hidden Objects in Structures	The subVI "Error Cluster From Error Code.vi" is not entirely within the visible bounds of its owner (Case Structure, "False" case).
Hidden Objects in Structures	This global variable (5056) is not entirely within the visible bounds of its owner (Case Structure, "False" case).
Hidden Objects in Structures	This constant is not entirely within the visible bounds of its owner (Case Structure, "False" case).


Testing Errors

(none)
```